pkg/service/model_build_listener.go

```diff
@@ -39,7 +65,7 @@ func (t *defaultModelBuildTask) buildListener(ctx context.Context, port corev1.S
 	return ls, nil
 }
 
-func (t *defaultModelBuildTask) buildListenerSpec(ctx context.Context, port corev1.ServicePort, cfg listenerConfig,
+func (t *defaultModelBuildTask) buildListenerSpec(ctx context.Context, port corev1.ServicePort, cfg *listenerConfig,
```

Comment:
I can't comment directly on it, but I think the logic for switching to `ProtocolTLS` immediately below probably needs to be specific to `ProtocolTCP` rather than `!ProtocolUDP` as I don't think turning a `TCP_UDP` port into a TLS port is a valid conversion. (Maybe it should fail if we get a `TCP_UDP` port where the TCP side would match the TLS configuration? (This might be the only place we use `port.Name`, from a brief check, which would also resolve the concern that the merged `TCP_UDP` port gains the name of the first-processed `ServicePort`, and loses the second's name.)